filebeat/_meta/common.full.p2.yml

```diff
@@ -82,6 +82,11 @@ filebeat.prospectors:
   # This is especially useful for multiline log messages which can get large.
   #max_bytes: 10485760
 
+  # Maximum depth for "**" patterns. This setting is disabled by default, i.e.,
```

Comment:
I somehow expect we will add more config options in the future. So I would suggest to make this a namespace with different config options below. Suggestion:

```
recursive_glob.enabled: false # set this to true to get the defaults for the below options
recursive_glob.depth: 16 # max depth of expanding **
recursive_glob.max_count: 1 # max number of ** in a single path
```
This could also solve the discussion about how many patterns we allow per path.

I named it `recursive_glob` because that what its doing from my perspective. Not sure what the user would look for when searching for such a config option. I had also `recursive_path` because we call it `paths` in the config above.